transport/src/main/java/io/netty/bootstrap/AbstractBootstrap.java

```diff
@@ -265,6 +262,7 @@ public ChannelFuture bind(InetAddress inetHost, int inetPort) {
      */
     public ChannelFuture bind(SocketAddress localAddress) {
         validate();
+        this.localAddress = localAddress;
```

Comment:
I think this is not correct... as its not a "builder" method which means it shouldn't change the bootstrap itself. @chrisvest thoughts ?

Reply:
I agree. The parameter is more like an override of a default. Those who want to know what address was bound in the end, will have to get the address via the ChannelFuture.

Reply:
Just reserve bind() without param will be more reasonable. But due to history issue, it can't be remove easily.

Reply:
So I think we should just close this as "won't fix"... @chrisvest agree ?

Reply:
For the address binding, yes. It still makes sense to have the exceptions unified in master, if they aren't already.

Reply:
reopen a pr to unify the exception message?

Reply:
reopen against master.